pandas/core/resample.py

```diff
@@ -83,14 +83,14 @@ class Resampler(BaseGroupBy, ShallowMixin):
         "offset",
     ]
 
-    def __init__(self, obj, groupby=None, axis=0, kind=None, **kwargs):
+    def __init__(self, obj, groupby=None, axis=0, kind=None, group_keys=True, **kwargs):
```

Comment:
why is `group_keys=False` just not the default here? e.g. why are we passing it at all

Reply:
cc @TomAugspurger 

I've never used resampler, so I don't have any opinion here on whether the API should expose group_keys. Removing group_keys from the API, all tests pass with the default `group_keys=False` except for `test_resample_group_keys`. This just tests that a warning is/is not raised in various cases and uses the API to do so.

Reply:
IIRC, Resampler internally creates a Grouper. We need to somehow pass `group_keys` to that so that there's no warning when using `.resample()`.

I don't recall if `.resample()` is actually affected by the values-dependent behavior. If it's not, then we can keep `group_keys` out of the public API and just hardcode the behavior we want. But if it does suffer from the same values-dependent behavior then we need to provide the keyword here.

I assume that `False` is the default for backwards compat.